extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/types/StringBloomFilterAggregatorColumnSelectorStrategy.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.bloom.types;
+
+import org.apache.druid.query.filter.BloomKFilter;
+import org.apache.druid.segment.DimensionSelector;
+
+import java.nio.ByteBuffer;
+
+public class StringBloomFilterAggregatorColumnSelectorStrategy
+    implements BloomFilterAggregatorColumnSelectorStrategy<DimensionSelector>
+{
+  @Override
+  public void add(DimensionSelector selector, BloomKFilter bloomFilter)
+  {
+    if (selector.getRow().size() > 1) {
+      selector.getRow().forEach(v -> {
+        String value = selector.lookupName(v);
```

Comment:
Javadocs aren't super clear and it isn't obvious to me yet looking at code, does `nameLookupPossibleInAdvance` indicate that indexes uniquely map to the same value across all segments? That would need to be true for this to work, and it would also require modifications to the `BloomDimFilter` implementation to be able to work with this as well.

I think the main use case of this aggregator right now is to be able to produce bloom filters from data in druid, which can be used as input to additional queries to filter other data in druid using `BloomDimFilter` which this extension originally introduced, so the goal is to have them operate in the same manner.

It definitely is expensive to do the hashing, it feels even apparent on the `BloomDimFilter` side of things, but I think this optimization is maybe out of scope of this PR

Reply:
> Javadocs aren't super clear and it isn't obvious to me yet looking at code, does nameLookupPossibleInAdvance indicate that indexes uniquely map to the same value across all segments? That would need to be true for this to work, and it would also require modifications to the BloomDimFilter implementation to be able to work with this as well.

Yes, map uniquely. Feel free to clarify that Javadoc right in this PR.

> I think the main use case of this aggregator right now is to be able to produce bloom filters from data in druid, which can be used as input to additional queries to filter other data in druid using BloomDimFilter which this extension originally introduced, so the goal is to have them operate in the same manner.

> It definitely is expensive to do the hashing, it feels even apparent on the BloomDimFilter side of things, but I think this optimization is maybe out of scope of this PR

OK, then please leave a comment somewhere with an explanation why the optimization is not applied.